test/step/nestedStep.js

```diff
@@ -463,4 +463,326 @@ describe('A step', () => {
+                            ]
+                        }]
+                })
+                assert.ok(st.auditTrail.elapsedTime > 0)
+            })
+
+            it('should doc', () => {
+                //given
+                const st = givenManyNestedSteps()
+                //when
+                const ret = st.doc()
+                //then
+                assert.deepStrictEqual(ret, {
+                    type: "step",
+                    description: undefined,
+                    steps: [
+                        {
+                            type: "step",
+                            description: 'step 1',
+                            steps: [
+                                { type: "step", description: 'step 1.1', steps: null }
+                            ]
+                        },
+                        {
+                            type: "step",
+                            description: 'step 2',
+                            steps: [
+                                { type: "step", description: 'step 2.1', steps: null },
+                                { type: "step", description: 'step 2.2', steps: null }
+                            ]
+                        },
+                        {
+                            type: "step",
+                            description: 'step 3',
+                            steps: [
+                                { type: "step", description: 'step 3.1', steps: null },
+                                { type: "step", description: 'step 3.2', steps: null },
+                                {
+                                    type: "step",
+                                    description: 'step 3.3',
+                                    steps: [
+                                        { type: "step", description: 'step 3.3.1', steps: null },
+                                        { type: "step", description: 'step 3.3.2', steps: null },
+                                        { type: "step", description: 'step 3.3.3', steps: null }
+                                    ]
+                                }
+                            ]
+                        }
+                    ]
+                })
+            })
+
+        })
+
+        context('returning a value', () => {
+
+            const givenNestedStepsWithNoReturn = () => {
+                const st = step({
+                    'step 1': step(() => { return Ok(1) }),
+                    'step 2': step((ctx) => { 
+                        ctx.stop()
+                        return 2 
+                    }),
+                    'step 3': step(() => { return [3] })
+                })
+                return st
+            }
+
+            it('should run', async () => {
+                //given
+                const st = givenNestedStepsWithNoReturn()
+                //when
+                const ret = await st.run()
+                //then
+                assert.ok(ret.isOk)
```

Comment:
It should check the returned value, right?

Reply:
Not in this case, return a value without Ok() make "UC result" test impossible, the return will be shown in the audit but can't be acess in "UC result " object. I repety this test (line 78) have sure `.stop()` don't change that behavior and in my undertand, the intuit of the original test is garant that one step returning a value don't broke the entiry UC